specs/0119-transactional-emails.md

```diff
@@ -0,0 +1,428 @@
+- Feature Name: Transactional Emails
+- Created: 2024-08-07
+- Associated Issue: #2076
+
+# Summary
+
+Implement the ability to send Pontoon notifications as an email, and to generate a number of automatically triggered transactional email messages to be sent to Pontoon users.
+
+# Motivation
+
+Users receive a number of useful and sometimes time-sensitive notifications within Pontoon. Some users have already indicated that they would like to receive notifications as emails, but we currently do not have the ability to send them. This feature would allow users to indicate which types of notifications they would like to receive as an email.
+
+In addition, this document contains a number of transactional emails to be sent to Pontoon users. The motivation for these are based on an [experiment](https://docs.google.com/document/d/1MY4H4jlxgYDmIsu8CW5q8VN0wo_nSWVCtRSdy119EGA/edit?usp=sharing) and work done in 2023 around transactional emails being sent to inactive users — these emails encouraged re-engagement of some lapsed/dormant users.
+
+Finally, some additional transactional emails are intended to complement upcoming initiatives for improving the onboarding experience for new contributors.
+
+# Feature explanation
+
+## Settings changes
+
+### Sending comments as email
+
+Users choose what notifications are sent to their email by an “Include in email digest” setting added under the existing “Notification subscriptions” section in settings. No email digest is sent if there are no notifications selected, otherwise an email digest is sent either daily or weekly based on the user’s selection from an “Email digest frequency” dropdown.
+
+This updates the “Notification subscriptions” section to include:
+
+* 2 columns with checkboxes for each notification type.  
+* Column 1 label: “Enable”  
+  * This maps to the current functionality — enabling this will enable notifications in Pontoon.  
+  * Specs / defaults should be preserved from current implementation.  
+* Column 2 label (default value: disabled): “Include in email digest”  
+  * When enabled, this type of notification will be included in an email sent to the user’s designated email address (either their login account, or the email address they define in the email preferences section.)  
+* Dropdown list (default value: Daily): “Email digest frequency”  
```

Comment:
Is Daily the right default here? I assume most of our active localizers don't localize every day, and daily emails will be much more annoying to them than weekly (which are also sent by SUMO).

Reply:
That's a good point. I was thinking the main use case here would be to get more timely notifications, but it makes sense to be more conservative with the default choice and making the less noisy version as default here.